__tests__/hooks/useDateFormatter.js

```diff
@@ -0,0 +1,20 @@
+import { timeouts } from '../constants.json';
```

Comment:
can you explain this one a little? I don't understand `formatDate` yet.

Reply:
This is for formatting timestamp into absolute short date/time.

Today, we are using `window.Intl`, which is not on all browsers (e.g. IE11).

Tomorrow, we are using `globalize`, a package for localizing everything defined by Unicode CLDR. That's why you are seeing `"12月23日 下午12:34"` now.

IMO, `globalize` is the de facto package for all localization need. It will be great if someone build polyfill out of this package. Reasons that `globalize` is great:

- Read from `cldr-data`, which compiles from official Unicode CLDR data and is considered very up-to-date
- CLDR data is huge, `globalize` can trim them down (a.k.a. precompile) to what you need. CLDR data is about 100 KB per language
- Single source of CLDR data, means, if I have two polyfills, they can use the same set of CLDR data when using `globalize`, that means smaller footprint